### PR TITLE
Fix pr helper

### PR DESCRIPTION
**Pull request by hogo6002**
Fix https://github.com/google/oss-fuzz/issues/10515:
1. Avoid repeated message: handle project with sub directory.
2. Simplify the email verification message
3. Link pull request without reference

### CODE REVIEW

File: .github/workflows/pr_helper.yml [hunk 1]
```diff
@@ -56,15 +56,3 @@ jobs:
               repo: context.repo.repo,
               body: '${{env.MESSAGE}}'
             })
-
```
Comment (jonathanmetzman) [inline .github/workflows/pr_helper.yml:59]: not sure we should get rid of it actually. let's leave for now.

File: infra/pr_helper.py [hunk 2]
```diff
@@ -151,14 +152,16 @@ def get_projects_path(self):
...
     for file in response.json():
       file_path = file['filename']
-      dir_path = os.path.dirname(file_path)
-      if dir_path is not None and dir_path.split(os.sep)[0] == 'projects':
-        projects_path.add(dir_path)
-    return list(set(projects_path))
+      dir_path = file_path.split(os.sep)
+      if len(dir_path) > 1 and dir_path[0] == 'projects':
+        projects_path.add(os.sep.join(dir_path[0:2]))
+    return list(projects_path)
```
Comment (jonathanmetzman) [inline infra/pr_helper.py:164]: Ha I guess my change did nothing. Thanks

**Review by jonathanmetzman — APPROVED** [review]
LGTM
Thanks for making these changes!

File: .github/workflows/pr_helper.yml (hunk 1 above)
Comment (oliverchang) [reply .github/workflows/pr_helper.yml:59]: +1. Was there some context for why we wanted to remove this?

File: infra/pr_helper.py [hunk 3]
```diff
@@ -75,7 +75,7 @@ def main():
   pr_author = github.get_pr_author()
   # Gets all modified projects path.
   projects_path = github.get_projects_path()
-  verified, email = github.get_author_email()
+  email = github.get_author_email()
```
Comment (oliverchang) [inline infra/pr_helper.py:78]: What's the reason for this change? Are we still verifying that the email is valid?

Comment (hogo6002) [reply infra/pr_helper.py:78]: We removed the verified check in the previous PR (https://github.com/google/oss-fuzz/pull/10519), so I removed this variable.  It seems that most emails are not verified, which is confusing for users (https://github.com/google/oss-fuzz/pull/10509#issuecomment-1588862254). If we don't want to remove the verified check, I can move the contact check after the PR commit history check.

Comment (oliverchang) [reply infra/pr_helper.py:78]: Ah I missed that part in my last review. I think as is this would also break because `github.get_author_email()` still returns a tuple? 

Can we add that verified check back? @jonathanmetzman any reasons for removing this? Otherwise this is trivial for contributors to spoof. We can simplify the output message here to just add a "(verified)" string if needed.

Comment (jonathanmetzman) [reply infra/pr_helper.py:78]: Ah I misunderstood this. Let's keep it. Sorry, Holly.

Comment (hogo6002) [reply infra/pr_helper.py]: No problem, added email verification back and simplified message.

File: infra/pr_helper.py [hunk 4]
```diff
@@ -100,9 +100,13 @@ def main():
     if email:
       if is_known_contributor(content_dict, email):
         # Checks if the email is verified.
-        message += (
-            f'{pr_author} is either the primary contact or is in the CCs list '
-            f'of [{project_path}]({project_url}).<br/>')
+        if verified:
+          message += (
+              f'{pr_author} (verified) is either the primary contact or '
```
Comment (oliverchang) [inline infra/pr_helper.py:105]: Can we just do something like this:

```pypi

verified_marker = ' (verified') if verified else ''
message = f'....{verified_marker}'
```

This way we avoid some duplication.

Comment (hogo6002) [reply infra/pr_helper.py:105]: For unverified email, should we continue checking PR commit history? Currently, we don't check history for verified email.

**Review by oliverchang — APPROVED** [review]
